src/doc/unstable-book/src/compiler-flags/source-based-code-coverage.md

```diff
@@ -0,0 +1,175 @@
+# `source-based-code-coverage`
+
+The feature request for this feature is: [#34701]
+
+The Major Change Proposal (MCP) for this feature is: [#278](https://github.com/rust-lang/compiler-team/issues/278)
+
+------------------------
+
+## Introduction
+
+The Rust compiler includes two code coverage implementations:
+
+* A GCC-compatible, gcov-based coverage implementation, enabled with [`-Zprofile`](profile.md), which operates on DebugInfo.
+* A source-based code coverage implementation, enabled with `-Zinstrument-coverage`, which uses LLVM's native coverage instrumentation to generate very precise coverage data.
+
+This document describes how to enable and use the LLVM instrumentation-based coverage, via the `-Zinstrument-coverage` compiler flag.
+
+## How it works
+
+When `-Zinstrument-coverage` is enabled, the Rust compiler enhances rust-based libraries and binaries by:
+
+* Automatically injecting calls to an LLVM intrinsic ([`llvm.instrprof.increment`]), at functions and branches in compiled code, to increment counters when conditional sections of code are executed.
+* Embedding additional information in the data section of each library and binary (using the [LLVM Code Coverage Mapping Format]), to define the code regions (start and end positions in the source code) being counted.
+
+When running a coverage-instrumented program, the counter values are written to a `profraw` file at program termination. LLVM bundles tools that read the counter results, combine those results with the coverage map (embedded in the program binary), and generate coverage reports in multiple formats.
+
+## Enable coverage profiling in the Rust compiler
+
+*IMPORTANT:* Rust's coverage profiling features may not be enabled, by default. To enable them, you may need to build a version of the Rust compiler with the `profiler` feature enabled.
+
+First, edit the `config.toml` file, and find the `profiler` feature entry. It may be commented out:
+
+```toml
+# Build the profiler runtime (required when compiling with options that depend
+# on this runtime, such as `-C profile-generate` or `-Z instrument-coverage`).
+#profiler = false
+```
+
+Uncomment the feature entry and set it to `true`:
+
+```toml
+profiler = true
+```
```

Comment:
```suggestion
First, edit the `config.toml` file, and find the `profiler` feature entry. Uncomment it and set it to `true`:

```toml
# Build the profiler runtime (required when compiling with options that depend
# on this runtime, such as `-C profile-generate` or `-Z instrument-coverage`).
profiler = true
```
```

The phrasing here is a bit verbose.